### PR TITLE
Modularise active promotions state

### LINKED ISSUE #42419: Modularise `activePromotions`
The `activePromotions` portion of the state tree needs to be modularised. See the [modularised state documentation](https://github.com/Automattic/wp-calypso/blob/master/docs/modularized-state.md) for more details.

### PR DESCRIPTION
**Pull request by sgomes**
This PR is one of many working on modularising state in Calypso. This one handles active promotions.

For more details on state modularisation, see the [modularised state documentation](https://github.com/Automattic/wp-calypso/blob/master/docs/modularized-state.md) and p4TIVU-9lM-p2.

Note: I added all the reviewers that GitHub suggested. Please feel free to ignore the review request or pull in someone else if you're not comfortable reviewing this PR. Thank you!

Fixes #42419.

#### Changes proposed in this Pull Request

* Modularise active promotions state
* Change some arrow function constants to plain functions

#### Testing instructions

* Install [Redux DevTools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en).
* Open DevTools and switch to the Redux tab.
* Open `/read` on the live branch or your local copy of this branch.
* Click `State` on the top right to ensure you see all state, and not just the diff for the last action: 
image
* Analyse the tree and note that there's no `activePromotions` key, even if you expand the list of keys. This indicates that the active promotions state hasn't been loaded yet.
* Click `My Sites` on the masterbar.
* Verify that a `activePromotions` key is added with the active promotions state. This indicates that the active promotions state has now been loaded.
* Verify that active promotion-related functionality works normally. This indicates that I didn't mess up.

### CODE REVIEW

**Comment by sgomes** [comment]
Note: the linting errors are expected, since I'm modifying the root reducer, which uses linting errors to point out unmodularised state in an attempt to prevent more of it from being created.

File: client/state/active-promotions/actions.js [hunk 1]
```diff
@@ -10,47 +9,46 @@ import {
...
+import 'state/active-promotions/init';
 
 /**
  * Action creator function: RECEIVE
  *
  * @param {Array} activePromotions - WordPress.com activePromotions list
  * @returns {object} action object
  */
-export const activePromotionsReceiveAction = ( activePromotions ) => {
+export function activePromotionsReceiveAction( activePromotions ) {
```
Comment (scinos) [inline client/state/active-promotions/actions.js:20]: Out of curiosity, what's the reason for this change?

**Review by scinos — APPROVED** [review]
* Lint errors are expected
* e2e errors seems unrelated to this change (problems with login and payments)
* `activePromotions` state is added when expected
* Everything seems to work fine

LGTM

File: client/state/active-promotions/actions.js (hunk 1 above)
Comment (sgomes) [reply client/state/active-promotions/actions.js:20]: It seems unnecessarily complicated to use an arrow function and store it in an exported `const`, when this can be a plain old function that doesn't need any transpilation.

An argument could be made for keeping it as a `const` + arrow function if it were using shorthand with an implicit return (no explicit `return` statement). Since this is already using brackets and a `return`, though, it just seems needlessly complex. A plain old function should be quicker to grok without doing a double take.

**Comment by sgomes** [comment]
Thanks for yet another review, @scinos! 👍
